src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/PropertiesShouldNotBeWriteOnlyTests.cs

```diff
@@ -9,12 +10,680 @@ public class PropertiesShouldNotBeWriteOnlyTests : DiagnosticAnalyzerTestBase
+            get { return field; }
+            internal set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_public_Read_internal_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        public string AccessibleProperty5
+        {
+            get { return field; }
+            internal set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_public_Read_protected_Write()
+        {
+            var code = @"
+using System;
+namespace DesignLibrary
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        public string AccessibleProperty6
+        {
+            get { return field; }
+            protected set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_public_override_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    protected string field;
+    public class DerivedClassWithReadableProperty : ClassWithReadableProperty
+    {
+        public override string ReadableProperty1
+        {
+            set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+// Cannot get below test to work, not sure where the issue is.
+//        [Fact]
+//        public void CSharp_CA1044Interface_Write()
+//        {
+//            var code = @"
+//using System;
+//namespace GoodPropertiesShouldNotBeWriteOnlyTests
+//{
+//    public interface IInterface
+//    {
+//        string InterfaceProperty
+//        {
+//            set;            
+//        }
+//    }
+//}";
+//            VerifyCSharp(code);
+//        }
+
+//        [Fact]
+//        public void CSharp_CA1044Interface_Write1()
+//        {
+//            var code = @"
+//using System;
+//namespace GoodPropertiesShouldNotBeWriteOnlyTests
+//{
+//    public class Class1 : IInterface
+//    {
+//        string IInterface.InterfaceProperty
+//        {
+//            set { }
+//        }
+//    }
+//}";
+//            VerifyCSharp(code);
+//        }
+
+//        [Fact]
+//        public void CSharp_CA1044Base_Write()
+//        {
+//            var code = @"
+//using System;
+//namespace GoodPropertiesShouldNotBeWriteOnlyTests
+//{
+//    public class Base
+//    {
+//        public virtual string BaseProperty
+//        {
+//            set { }
+//        }
+//    }
+//}";
+//            VerifyCSharp(code);
+//        }
+
+        [Fact]
+        public void CSharp_CA1044Base_Write1()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class Derived : Base
```

Comment:
This code snippet will have errors because Base is not defined in this snippet. You should combine this snippet with the previous test case (with the virtual method) into one test.